src/httpserver/qabstracthttpserver.cpp

```diff
@@ -127,8 +129,12 @@ void QAbstractHttpServerPrivate::handleReadyRead(QTcpSocket *socket,
         Q_EMIT q->missingHandler(*request, socket);
 }
 
+QIODevice* QAbstractHttpServer::createBodyDevice(const QHttpServerRequest& /*request*/) {
+    return new QBuffer();
+}
+
 QAbstractHttpServer::QAbstractHttpServer(QObject *parent)
-    : QAbstractHttpServer(*new QAbstractHttpServerPrivate, parent)
+    : QAbstractHttpServer(*new QAbstractHttpServerPrivate(this), parent)
```

Comment:
This is some kind of mutual dependency. Generally the symptom of a design problem.
The destructor might be tricky :s